### PR TITLE
feat: Allow fractional uakt for settlement

### PR DESCRIPTION
**Pull request by hydrogen18**
Stuff I changed

1. For escrow accounts, use a decimal coin
2. For bid pricing and deployment max price, use a decimal coin
3. Rework some of the settlement code, since with decimal types it isn't really as complex
4. Migrate all of our types to v1beta2 
5. Keep the v1beta1 code for now, in a package called `typesv1beta` where appropriate
6. Make some of the logic around provider attribute types code use an interface instead, so they can run on any version
6. Add a migration that reads the binary values, parses them, and writes them back as v1beta2 version
7. In our examples, lower the proposal time down to 15 seconds so we can actually do proposals when testing

### CODE REVIEW

**Comment by boz** [comment]
bump

**Comment by boz** [comment]
refs #1174

File: proto/akash/audit/v1beta2/query.proto [hunk 1]
```diff
@@ -0,0 +1,79 @@
+syntax = "proto3";
```
Comment (boz) [inline proto/akash/audit/v1beta2/query.proto:1]: We can use a mix of versions - not everything necessarily needs to move to `v1beta2`.

File: provider/bidengine/pricing.go [hunk 2]
```diff
@@ -108,34 +108,40 @@ func (fp scalePricing) CalculatePrice(_ context.Context, gspec *dtypes.GroupSpec
 
 	endpointTotal = endpointTotal.Mul(fp.endpointScale)
 
-	maxAllowedValue := decimal.NewFromInt(math.MaxInt64)
+	decimal.NewFromInt(math.MaxInt64)
```
Comment (boz) [inline provider/bidengine/pricing.go:111]: :scissors:

File: x/escrow/keeper/keeper.go [hunk 3]
```diff
@@ -392,11 +418,13 @@ func (k *keeper) doAccountSettle(ctx sdk.Context, id types.AccountID) (types.Acc
 	account, payments, amountRemaining = accountSettleDistributeWeighted(
 		account, payments, blockRate, amountRemaining)
 
+	/**
```
Comment (boz) [inline x/escrow/keeper/keeper.go:421]: :scissors:

File: x/escrow/keeper/keeper.go [hunk 4]
```diff
@@ -530,18 +562,19 @@ func accountSettleFullblocks(
...
 
 	account.Transferred = account.Transferred.Add(transferred)
 	account.Balance = account.Balance.Sub(transferred)
 
 	remaining := account.Balance
 	overdrawn := true
 	if numFullBlocks.Equal(heightDelta) {
-		remaining.Amount = sdk.ZeroInt()
+		remaining.Amount = sdk.ZeroDec()
 		overdrawn = false
```
Comment (boz) [inline x/escrow/keeper/keeper.go:578]: does this still work if we're using `TruncateInt()` when calculating `numFullBlocks`?

File: x/escrow/keeper/keeper.go [hunk 5]
```diff
@@ -550,31 +583,32 @@ func accountSettleFullblocks(
...
-	actualTransferred := sdk.ZeroInt()
+	actualTransferred := sdk.ZeroDec()
 
 	// distribute remaining balance weighted by rate
+
 	for idx := range payments {
 		payment := payments[idx]
 		// amount := (rate / blockrate) * remaining
 		amount := amountRemaining.Amount.
 			Mul(payment.Rate.Amount).
```
Comment (boz) [inline x/escrow/keeper/keeper.go:603]: uh... does this mutate `amountRemaining.Amount`?

**Review by boz — COMMENTED** [review]
Looking good.  Sorry for the delay.  Is there any chance that we can only upgrade the package versions for proto defs that need to be updated for this?

Note: we'll need to revive the `mainnet/main` branch and release from there before merging this.

**Comment by hydrogen18** [comment]
> Looking good. Sorry for the delay. Is there any chance that we can only upgrade the package versions for proto defs that need to be updated for this?
> 
> Note: we'll need to revive the `mainnet/main` branch and release from there before merging this.

If we stagger versions like that, how would we know what versions of what types to use?

**Comment by boz** [comment]
> If we stagger versions like that, how would we know what versions of what types to use?

```go
import xv1beta1 "akash/proto/v1beta1"
import yv1beta2 "akash/proto/v1beta2"
```

File: x/escrow/keeper/keeper.go (hunk 5 above)
Comment (hydrogen18) [reply x/escrow/keeper/keeper.go:603]: I don't actually know, I'll have to look at the implementation.

**Comment by hydrogen18** [comment]
Ok I will work on figuring out how to stagger the versions then

**Comment by boz** [comment]
superseded by #1394

**Pull request closed without merging**